shm/src/Item.cpp

```diff
@@ -0,0 +1,48 @@
+#include "shm/inc/Item.hpp"
+
+Item::Item(const std::string& name, size_t amount, size_t basePrice, Rarity rarity)
+    : Cargo(name, amount, basePrice), rarity_(rarity)
+{}
+
+bool Item::operator==(const Cargo& cargo) const {
+    if (typeid(cargo) == typeid(Item)) {
+        const Item* item = static_cast<const Item*>(&cargo);
+        return name_ == item->getName() && amount_ == item->getAmount() && basePrice_ == item->getBasePrice() &&
+               rarity_ == item->getRarity();
```

Comment:
Im not sure if we need compare amout, amout don't need be the same.

Reply:
It depends but I'll remove it for now.